core/src/main/java/hudson/model/RootAction.java

```diff
@@ -50,4 +50,17 @@ public interface RootAction extends Action, ExtensionPoint {
     default @CheckForNull Badge getBadge() {
         return null;
     }
+
+    /**
+     * Identifies if the action as a primary action.
+     * Primary actions may be handled differently in the UI (for existence by always showing on the header rather than in an actions dropdown).
+     * In almost all cases this should return {@code false} which is the default.
+     * Examples of a Primary action would be where actions would be used regularly or they would need to be prominent to convey some status.
+     * A concrete example would be an action that provides for Help with a submenu containing various different links would be a prime candidate for this to return true.
+     * If your action has the possibility of a {@link #getBadge() badge} then this is another indicator that the action should be a primary action (or otherwise refactored!)
```

Comment:
```suggestion
     * If your action has the possibility of a {@link #getBadge() badge} then this is another indicator that the action should be a primary action (or otherwise restructured!)
```